pkg/kp/consulutil/session_controller.go

```diff
@@ -34,8 +40,29 @@ func SessionManager(
 	done chan struct{},
 	logger logging.Logger,
 ) {
+	prng := randseed.NewRand()
+	minDelay := time.Duration(*SessionRetrySeconds) * time.Second
+	maxDelay := time.Duration(*SessionMaxRetrySeconds) * time.Second
+	useDelay := false
+	var delay time.Duration
+
 	logger.NoFields().Info("session manager: starting up")
 	for {
+		if useDelay {
+			// Normalize timeout range
+			if delay < minDelay {
+				delay = minDelay
+			} else if delay > maxDelay {
+				delay = maxDelay
+			}
+			select {
+			case <-time.After(time.Duration(prng.Int63n(int64(delay)))):
```

Comment:
Should this be `time.Duration(minDelay + prng.Int63n(int64(delay-minDelay)))`? Really, I'm poking at the meaning of "minDelay" when it could actually wait less time than this value implies.